halyard/halyard-config/src/test/java/com/netflix/spinnaker/halyard/config/validate/v1/providers/dockerRegistry/DockerRegistryAccountValidatorTest.java

```diff
@@ -0,0 +1,125 @@
+package com.netflix.spinnaker.halyard.config.validate.v1.providers.dockerRegistry;
+
+import static com.github.tomakehurst.wiremock.client.WireMock.urlEqualTo;
+import static com.github.tomakehurst.wiremock.core.WireMockConfiguration.wireMockConfig;
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.github.tomakehurst.wiremock.client.WireMock;
+import com.github.tomakehurst.wiremock.junit5.WireMockExtension;
+import com.netflix.spinnaker.clouddriver.docker.registry.api.v2.auth.DockerBearerToken;
+import com.netflix.spinnaker.clouddriver.docker.registry.api.v2.client.DockerRegistryCatalog;
+import com.netflix.spinnaker.halyard.config.config.v1.HalconfigDirectoryStructure;
+import com.netflix.spinnaker.halyard.config.model.v1.providers.dockerRegistry.DockerRegistryAccount;
+import com.netflix.spinnaker.halyard.config.problem.v1.ConfigProblemSetBuilder;
+import com.netflix.spinnaker.halyard.config.services.v1.FileService;
+import com.netflix.spinnaker.halyard.core.problem.v1.Problem;
+import com.netflix.spinnaker.halyard.core.secrets.v1.SecretSessionManager;
+import com.netflix.spinnaker.kork.configserver.CloudConfigResourceService;
+import com.netflix.spinnaker.kork.configserver.ConfigFileService;
+import com.netflix.spinnaker.kork.secrets.SecretEngineRegistry;
+import com.netflix.spinnaker.kork.secrets.SecretManager;
+import java.util.List;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+@SpringBootTest(
+    classes = {
+      SecretSessionManager.class,
+      SecretManager.class,
+      SecretEngineRegistry.class,
+      FileService.class,
+      ConfigFileService.class,
+      CloudConfigResourceService.class,
+      DockerRegistryAccountValidatorTest.TestConfig.class
+    })
+public class DockerRegistryAccountValidatorTest {
+  @Autowired ApplicationContext context;
+
+  @Autowired SecretSessionManager secretSessionManager;
+
+  @Autowired FileService fileService;
+
+  @Autowired DockerRegistryAccountValidator validator;
+
+  @RegisterExtension
+  static WireMockExtension wmDocker =
+      WireMockExtension.newInstance().options(wireMockConfig().dynamicPort()).build();
+
+  ConfigProblemSetBuilder configProblemSetBuilder = new ConfigProblemSetBuilder(context);
```

Comment:
Seems safer to move this to the test method, since that's the only place it's used.  Since context is autowired, it may not be populated properly here.

Reply:
Done